packages/desktop/electron/lib/ledger.js

```diff
@@ -10,34 +10,91 @@ const TransportNodeHid = require('@ledgerhq/hw-transport-node-hid').default
 const AppEth = require('@ledgerhq/hw-app-eth').default
 const { listen } = require('@ledgerhq/logs')
 
+const { Chain, Common } = require('@ethereumjs/common')
+const { RLP } = require('@ethereumjs/rlp')
+const { Transaction } = require('@ethereumjs/tx')
+const { bufArrToArr } = require('@ethereumjs/util')
+
+let transport
+
 process.parentPort.on('message', async (message) => {
     try {
+        await openTransport()
+
         let data
         switch (message.data.method) {
             case 'generate-evm-address': {
                 data = await getEvmAddress(...message.data.parameters)
                 break
             }
+            case 'sign-evm-transaction': {
+                data = await signTransactionData(...message.data.parameters)
+                break
+            }
             default:
                 break
         }
+
+        await closeTransport()
+
         process.parentPort.postMessage({ data })
     } catch (error) {
         process.parentPort.postMessage({ error })
     }
 })
 
+async function openTransport() {
+    if (!transport) {
+        transport = await TransportNodeHid.open('')
+        listen((log) => {
+            process.parentPort.postMessage({ data: log })
+        })
+    }
+}
+
+async function closeTransport() {
+    if (transport) {
+        await transport.close()
+        transport = undefined
+    }
+}
+
 async function getEvmAddress(coinType, accountIndex, verify) {
-    const transport = await TransportNodeHid.open('')
-    listen((log) => {
-        process.parentPort.postMessage({ data: log })
-    })
     const appEth = new AppEth(transport)
     const data = await appEth.getAddress(buildBip32Path(coinType, accountIndex))
-    await transport.close()
+
     return { evmAddress: data.address, coinType, accountIndex }
 }
 
 function buildBip32Path(coinType, accountIndex) {
     return `44'/${coinType}'/${accountIndex}'/0/0`
 }
+
+async function signTransactionData(data, coinType, accountIndex) {
+    const appEth = new AppEth(transport)
+    const common = new Common({ chain: Chain.Sepolia })
```

Comment:
use shimmerEVM

Reply:
I removed showing the Chain Id, here because it requires more work. You have to config a custom chain and we're lacking some data.